### PR TITLE
[MRG+1] Issue 146

### PR DESCRIPTION
**Pull request by tgsmith61591**
# Description

Adds `**sarimax_kwargs` to the `ARIMA` constructor and `auto_arima` function. This addresses #146. While I was at it, I changed the `plot_diagnostics` unit tests to use `unittest.mock.patch` to make them less brittle. 

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

- Added a new test that shows that `simple_differencing=True` can fix an `auto_arima`
- Changed `plot_diagnostic` tests to use mocking

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: pmdarima/arima/_auto_solvers.py [hunk 1]
```diff
@@ -56,6 +57,7 @@ def __init__(self, y, xreg, start_params, trend, method, transparams,
         self.scoring_args = scoring_args
         self.information_criterion = information_criterion
         self.with_intercept = with_intercept
+        self.kwargs = dict() if not kwargs else kwargs
```
Comment (aaronreidsmith) [inline pmdarima/arima/_auto_solvers.py]: Is there a reason to use `dict()` vs `{}`? I know `{}` is faster (https://stackoverflow.com/a/5790954)

File: pmdarima/arima/auto.py [hunk 2]
```diff
@@ -96,6 +97,7 @@ def __init__(self, start_p=2, d=None, start_q=2, max_p=5,
         self.scoring = scoring
         self.scoring_args = scoring_args
         self.with_intercept = with_intercept
+        self.sarimax_kwargs = dict() if not sarimax_kwargs else sarimax_kwargs
```
Comment (aaronreidsmith) [inline pmdarima/arima/auto.py]: Same comment regarding `dict()` vs `{}` (just respond to one, just wanted to point both out)

File: pmdarima/arima/auto.py [hunk 3]
```diff
@@ -247,6 +251,8 @@ def auto_arima(y, exogenous=None, start_p=2, d=None, start_q=2, max_p=5,
                                  force_all_finite=True))  # type: np.ndarray
     n_samples = y.shape[0]
 
+    sarimax_kwargs = dict() if not sarimax_kwargs else sarimax_kwargs
```
Comment (aaronreidsmith) [inline pmdarima/arima/auto.py]: Same comment (sorry for spam 😂)

File: pmdarima/arima/_auto_solvers.py (hunk 1 above)
Comment (tgsmith61591) [reply pmdarima/arima/_auto_solvers.py]: `{}` can be a [set literal](https://stackoverflow.com/questions/36674083/why-is-it-possible-to-replace-sometimes-set-with) 😬

Comment (aaronreidsmith) [reply pmdarima/arima/_auto_solvers.py]: The way I understand it, Python will _always_ interpret `{}` as an empty dictionary. So you end up with the following:

```python
example1 = {}  # Empty dict
example2 = {1: 'test'}  # Dict with one item
example3 = set()  # Empty set
example4 = {1, 2, 3}  # Set with 3 items
```
[The official docs even define `{}` as an empty dict.](https://docs.python.org/3.7/library/stdtypes.html#truth-value-testing) The _only_ definition of an empty set is `set()`

Am I off base? I don't think the fraction of a second is a huge deal, but we _do_ get a lot of questions about performance, so I figured I would bring it up!

**Pull request merged**